### PR TITLE
Fluentd improvements to lessen likelihood of buffers filling up and hanging

### PR DESCRIPTION
**Pull request by a-robinson**
Context in https://github.com/kubernetes/kubernetes/issues/19405#issuecomment-175296741

tl;dr this should make fluentd handle high logging rates much better than a few users have found it to handle them

I'd like to squeeze this into 1.1.6 to get it out to GKE users sooner, so a quick review would be appreciated :)

@mr-salty @vishh @roberthbailey

### CODE REVIEW

**Comment by roberthbailey** [comment]
LGTM.

File: cluster/addons/fluentd-elasticsearch/fluentd-es-image/td-agent.conf [hunk 1]
```diff
@@ -186,9 +191,9 @@
    port 9200
    logstash_format true
    # Set the chunk limit the same as for fluentd-gcp.
-   buffer_chunk_limit 512K
-   # Cap buffer memory usage to 512KB/chunk * 128 chunks = 65 MB
-   buffer_queue_limit 128
+   buffer_chunk_limit 2M
+   # Cap buffer memory usage to 2MB/chunk * 32 chunks = 65 MB
```
Comment (mr-salty) [inline cluster/addons/fluentd-elasticsearch/fluentd-es-image/td-agent.conf]: funny math :)  I'd probably write 2MiB ... = 64MiB

**Comment by mr-salty** [comment]
LGTM

File: cluster/addons/fluentd-elasticsearch/fluentd-es-image/td-agent.conf (hunk 1 above)
Comment (a-robinson) [inline cluster/addons/fluentd-elasticsearch/fluentd-es-image/td-agent.conf]: done

**Pull request merged**